### PR TITLE
added release notes for 1.2.4

### PR DESCRIPTION
**Pull request by abhinavGupta16**
Signed-off-by: Abhinav Gupta <abhng@amazon.com>

### Description
Added release Notes for `1.2.4`
 
### Issues Resolved
Added release notes for `1.2.4`
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: release-notes/opensearch-release-notes-1.2.4.md [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+### OpenSearch Alerting
+
+* Bumps cron-utils version to 9.1.6. ([#205](https://github.com/opensearch-project/alerting/pull/274))
+
+### OpenSearch Anomaly-Detection
+
+* upgrade gson to 2.8.9 ([#354](https://github.com/opensearch-project/anomaly-detection/pull/354))
+* fix missing commons-lang3;update job scheduler plugin zip ([#364](https://github.com/opensearch-project/anomaly-detection/pull/364))
+* remove jcenter dependency ([#361](https://github.com/opensearch-project/anomaly-detection/pull/361))
```
Comment (peterzhuamazon) [inline release-notes/opensearch-release-notes-1.2.4.md:16]: Capitalize these sentences.

File: release-notes/opensearch-release-notes-1.2.4.md [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+
+### Opensearch SQL
+
+* Bump log4j to 2.17.1 and plugin to 1.2.4 ([#356](https://github.com/opensearch-project/sql/pull/356))
+* Bump gson and json-schema ([#370](https://github.com/opensearch-project/sql/pull/370))
+    
+### Opensearch Observability
+
+* Backporting CVE updates [#369](https://github.com/opensearch-project/observability/pull/369)
+* Incremented version to 1.2.4. [#362](https://github.com/opensearch-project/observability/pull/362)
```
Comment (peterzhuamazon) [inline release-notes/opensearch-release-notes-1.2.4.md:39]: Add parenthesis to the PR links.

**Review by peterzhuamazon — APPROVED** [review]
Thanks.

**Review by bbarani — APPROVED** [review]

File: release-notes/opensearch-release-notes-1.2.4.md [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+### OpenSearch Anomaly-Detection
+
+* Upgrade gson to 2.8.9 ([#354](https://github.com/opensearch-project/anomaly-detection/pull/354))
+* Fix missing commons-lang3;update job scheduler plugin zip ([#364](https://github.com/opensearch-project/anomaly-detection/pull/364))
+* Remove jcenter dependency ([#361](https://github.com/opensearch-project/anomaly-detection/pull/361))
+
+### OpenSearch Reports Scheduler
+
+* Update CVEs for 1.2 branch ([#270](https://github.com/opensearch-project/dashboards-reports/pull/270))
+* update yarn lock for cve fixes ([#273](https://github.com/opensearch-project/dashboards-reports/pull/273))
```
Comment (gaiksaya) [inline release-notes/opensearch-release-notes-1.2.4.md:21]: Missed one

**Pull request merged**